src/main/java/spireMapOverhaul/zones/manasurge/events/ManaCycloneEvent.java

```diff
@@ -0,0 +1,154 @@
+                        case 2:
+                            if (AbstractDungeon.player.hasRelic("Bottled Lightning")) {
+                                this.relicMetric = RelicLibrary.getRelic("Circlet").makeCopy();
+                            } else {
+                                this.relicMetric = RelicLibrary.getRelic("Bottled Lightning").makeCopy();
+                            }
+                            AbstractDungeon.getCurrRoom().spawnRelicAndObtain((float)(Settings.WIDTH / 2), (float)(Settings.HEIGHT / 2), this.relicMetric);
+                            break;
+                        case 3:
+                            if (AbstractDungeon.player.hasRelic("Frozen Egg 2")) {
+                                this.relicMetric = RelicLibrary.getRelic("Circlet").makeCopy();
+                            } else {
+                                this.relicMetric = RelicLibrary.getRelic("Frozen Egg 2").makeCopy();
+                            }
+                            AbstractDungeon.getCurrRoom().spawnRelicAndObtain((float)(Settings.WIDTH / 2), (float)(Settings.HEIGHT / 2), this.relicMetric);
+                            break;
+                        case 4:
+                            if (AbstractDungeon.player.hasRelic("Molten Egg 2")) {
+                                this.relicMetric = RelicLibrary.getRelic("Circlet").makeCopy();
+                            } else {
+                                this.relicMetric = RelicLibrary.getRelic("Molten Egg 2").makeCopy();
+                            }
+                            AbstractDungeon.getCurrRoom().spawnRelicAndObtain((float)(Settings.WIDTH / 2), (float)(Settings.HEIGHT / 2), this.relicMetric);
+                            break;
+                        case 5:
+                            if (AbstractDungeon.player.hasRelic("Toxic Egg 2")) {
+                                this.relicMetric = RelicLibrary.getRelic("Circlet").makeCopy();
+                            } else {
+                                this.relicMetric = RelicLibrary.getRelic("Toxic Egg 2").makeCopy();
+                            }
+                            AbstractDungeon.getCurrRoom().spawnRelicAndObtain((float)(Settings.WIDTH / 2), (float)(Settings.HEIGHT / 2), this.relicMetric);
+                            break;
+                    }
+                    CardGroup filteredCards = new CardGroup(CardGroup.CardGroupType.UNSPECIFIED);
+                    AbstractDungeon.player.masterDeck.group.stream()
+                            .filter(card -> !ManaSurgeZone.hasManaSurgeModifier(card) &&
+                                    card.cost != -2 &&
+                                    card.type != AbstractCard.CardType.CURSE &&
+                                    card.type != AbstractCard.CardType.STATUS)
+                            .forEach(filteredCards::addToTop);
+                    AbstractCard randomCard1 = filteredCards.getRandomCard(true);
```

Comment:
It's fine but this could've been a simple for i loop

Reply:
wouldn't that have the chance to roll the same card twice?

(if you mean this)
`for (int i = 0; i < 2; i++) {
    randomCard = filteredCards.getRandomCard(true);
    if (randomCard != null) {
        ManaSurgeZone.applyPermanentNegativeModifier(randomCard);
        AbstractDungeon.effectList.add(new ShowCardBrieflyEffect(randomCard.makeStatEquivalentCopy()));
        filteredCards.removeCard(randomCard);
    }
}`

Reply:
it did not space itself nicely i apologise

Reply:
i haven't changed it for now bc I'm worried I'll do something wrong but let me know if you'd like it changed